superset-frontend/src/dashboard/components/PropertiesModal/PropertiesModal.test.tsx

```diff
@@ -0,0 +1,319 @@
+import PropertiesModal from '.';
+
+const spyIsFeatureEnabled = jest.spyOn(FF, 'isFeatureEnabled');
+const spyColorSchemeControlWrapper = jest.spyOn(
+  ColorSchemeControlWrapper,
+  'default',
+);
+
+spyColorSchemeControlWrapper.mockImplementation(
+  () => (<div>ColorSchemeControlWrapper</div>) as any,
+);
+
+fetchMock.get(
+  'http://localhost/api/v1/dashboard/related/roles?q=(filter:%27%27)',
+  {
+    body: {
+      count: 6,
+      result: [
+        {
+          text: 'Admin',
+          value: 1,
+        },
+        {
+          text: 'Alpha',
+          value: 3,
+        },
+        {
+          text: 'Gamma',
+          value: 4,
+        },
+        {
+          text: 'granter',
+          value: 5,
+        },
+        {
+          text: 'Public',
+          value: 2,
+        },
+        {
+          text: 'sql_lab',
+          value: 6,
+        },
+      ],
+    },
+  },
+);
+
+fetchMock.get(
+  'http://localhost/api/v1/dashboard/related/owners?q=(filter:%27%27)',
+  {
+    body: {
+      count: 1,
+      result: [
+        {
+          text: 'Superset Admin',
+          value: 1,
+        },
+      ],
+    },
+  },
+);
+
+fetchMock.get('http://localhost/api/v1/dashboard/26', {
+  body: {
+    result: {
+      changed_by: null,
+      changed_by_name: '',
+      changed_by_url: '',
+      changed_on: '2021-03-30T19:30:14.020942',
+      charts: [
+        'Vaccine Candidates per Country & Stage',
+        'Vaccine Candidates per Country',
+        'Vaccine Candidates per Country',
+        'Vaccine Candidates per Approach & Stage',
+        'Vaccine Candidates per Phase',
+        'Vaccine Candidates per Phase',
+        'Vaccine Candidates per Country & Stage',
+        'Filtering Vaccines',
+      ],
+      css: '',
+      dashboard_title: 'COVID Vaccine Dashboard',
+      id: 26,
+      json_metadata:
+        '{"timed_refresh_immune_slices": [], "expanded_slices": {}, "refresh_frequency": 0, "default_filters": "{}", "color_scheme": "supersetColors", "label_colors": {"0": "#D3B3DA", "1": "#9EE5E5", "0. Pre-clinical": "#1FA8C9", "2. Phase II or Combined I/II": "#454E7C", "1. Phase I": "#5AC189", "3. Phase III": "#FF7F44", "4. Authorized": "#666666", "root": "#1FA8C9", "Protein subunit": "#454E7C", "Phase II": "#5AC189", "Pre-clinical": "#FF7F44", "Phase III": "#666666", "Phase I": "#E04355", "Phase I/II": "#FCC700", "Inactivated virus": "#A868B7", "Virus-like particle": "#3CCCCB", "Replicating bacterial vector": "#A38F79", "DNA-based": "#8FD3E4", "RNA-based vaccine": "#A1A6BD", "Authorized": "#ACE1C4", "Non-replicating viral vector": "#FEC0A1", "Replicating viral vector": "#B2B2B2", "Unknown": "#EFA1AA", "Live attenuated virus": "#FDE380", "COUNT(*)": "#D1C6BC"}, "filter_scopes": {"358": {"Country_Name": {"scope": ["ROOT_ID"], "immune": []}, "Product_Category": {"scope": ["ROOT_ID"], "immune": []}, "Clinical Stage": {"scope": ["ROOT_ID"], "immune": []}}}}',
+      owners: [],
+      position_json:
```

Comment:
I wonder if this can be in another file or be used as mock file somewhere? I have sense that this could be reused in multiple tests.

Reply:
This is a possibility, but given the lack of typing and organization it is very hard to know where these calls are being reused. I believe that this would be a future improvement, but not within the scope of this PR.

Sharing something we don't know where and when it will be used, in the current state of the project seems to me more harmful than good.

And we also have the problem that the reviewers of the other codes may not even know about the existence of these mocks... I agree that we need to centralize these things, but I think that now is not the time to do that.

Reply:
Agreed... we can put a pass on our roadmap to audit for reusable mock data, and make it a more systemic process/library, I could see that being a huge timesaver.